aea/cli/core.py

```diff
@@ -193,6 +197,100 @@ def add_key(ctx: Context, type_, file):
     )
 
 
+@cli.command()
+@click.argument(
+    "type_",
+    metavar="TYPE",
+    type=click.Choice(
+        [DefaultCrypto.identifier, FetchAICrypto.identifier, EthereumCrypto.identifier]
+    ),
+    required=True,
+)
+@pass_ctx
+def get_address(ctx: Context, type_):
+    """Get the address associated with the private key."""
+    try_to_load_agent_config(ctx)
+
+    _verify_or_create_private_keys(ctx)
+    private_key_paths = dict(
+        [
+            (identifier, config.path)
+            for identifier, config in ctx.agent_config.private_key_paths.read_all()
+        ]
+    )
+    try:
+        wallet = Wallet(private_key_paths)
+        address = wallet.addresses[type_]
+        logger.info("Your address is: {}".format(address))
+    except ValueError as e:  # pragma: no cover
+        logger.error(str(e))  # pragma: no cover
+
+
+@cli.command()
+@click.argument(
+    "type_",
+    metavar="TYPE",
+    type=click.Choice([FetchAICrypto.identifier, EthereumCrypto.identifier]),
+    required=True,
+)
+@pass_ctx
+def get_wealth(ctx: Context, type_):
+    """Get the wealth associated with the private key."""
+    try_to_load_agent_config(ctx)
+
+    _verify_or_create_private_keys(ctx)
+    private_key_paths = dict(
+        [
+            (identifier, config.path)
+            for identifier, config in ctx.agent_config.private_key_paths.read_all()
+        ]
+    )
+    wallet = Wallet(private_key_paths)
+    try:
+        _verify_ledger_apis_access()
+        ledger_api_configs = dict(
+            [
+                (identifier, cast(List[Union[str, int]], list(config.values())))
+                for identifier, config in ctx.agent_config.ledger_apis.read_all()
+            ]
+        )
+        ledger_apis = LedgerApis(ledger_api_configs, ctx.agent_config.default_ledger)
+
+        address = wallet.addresses[type_]
+        balance = ledger_apis.token_balance(type_, address)
+        logger.info("Your wealth for address {} is: {}".format(address, balance))
+    except (AssertionError, ValueError) as e:  # pragma: no cover
+        logger.error(str(e))  # pragma: no cover
+
+
+@cli.command()
+@click.argument(
+    "type_",
+    metavar="TYPE",
+    type=click.Choice([FetchAICrypto.identifier, EthereumCrypto.identifier]),
+    required=True,
+)
+@pass_ctx
+def generate_wealth(ctx: Context, type_):
```

Comment:
I'd like an option to specify the amount of wealth to generate

Reply:
That's not possible. Faucets generate an amount determined by the faucet provider, that's industry standard.

Reply:
@DavidMinarsch what about an optional argument to wait until the confirmation?
e.g. `--sync`

Reply:
Good point, will open an issue for that!

Reply:
https://github.com/fetchai/agents-aea/issues/663